src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -61,6 +63,9 @@ public static Task<RazorLanguageServer> CreateAsync(Stream input, Stream output,
             Serializer.Instance.Settings.Converters.Add(SemanticTokensOrSemanticTokensEditsConverter.Instance);
             Serializer.Instance.JsonSerializer.Converters.RegisterRazorConverters();
 
+            // Custom ClientCapabilities deserializer to extract experimental capabilities
+            Serializer.Instance.JsonSerializer.Converters.Add(ExtendableClientCapabilitiesJsonConverter.Instance);
```

Comment:
Hmmmm, if we start sharing versions of O# with WTE this static json serializer business may become troublesome. Food for thought. /cc @ToddGrun @jimmylewis @alexgav

Reply:
From what I could tell this was the only way to access the VS experimental capabilities through O#, as VS doesn't use the `ClientCapabilities.Experimental` [field](https://github.com/OmniSharp/csharp-language-server-protocol/blob/6371386ee90dde8a1c419f3dc8fc94e0c1bc35f0/src/Protocol/Client/Capabilities/ClientCapabilities.cs#L26), but rather adds top level `ClientCapabilities.XYZExperimental` variables. 

Is there a different technique used by WTE?